lib/temporal/workflow/context.rb

```diff
@@ -203,7 +203,8 @@ def continue_as_new(*input, **args)
           input: input,
           timeouts: execution_options.timeouts,
           retry_policy: execution_options.retry_policy,
-          headers: execution_options.headers
+          headers: execution_options.headers,
+          memo: execution_options.memo,
```

Comment:
This is also new in this latest iteration. The ensures the memo is copied in a continue-as-new.